torch/_meta_registrations.py

```diff
@@ -1017,6 +1155,41 @@ def meta_like(self, *args, **kwargs):
     return aten.empty_like.default(self, **kwargs)
 
 
+@register_meta(aten.select.int)
+def meta_select(self, dim, index):
+    ndim = self.dim();
+    if ndim == 0:
+        raise IndexError("select() cannot be applied to a 0-dim tensor.")
+
+    dim = dim if dim >= 0 else dim + ndim
+    size = self.size(dim)
+
+    if (-index > size) or index >= size:
+        raise IndexError(f"select(): index {index} out of range for tensor of size "
+                         f"{self.size()} at dimension {dim}")
+
+    index = index if index >= 0 else index + size
+
+    new_size = list(self.size())
+    new_stride = list(self.stride())
+
+    new_storage_offset = self.storage_offset() + index * new_stride[dim]
+    del new_size[dim]
+    del new_stride[dim]
+
+    return self.as_strided(new_size, new_stride, new_storage_offset)
```

Comment:
With this meta implementation, how come the C++ version also needed to be symint-ified?

Reply:
Not symintifying the C++ version reduces land risk, as the code running in prod is unchanged. But outside of that I don't really care; C++ symints will certainly run faster.